Tests/StructuredQueriesTests/InsertTests.swift

```diff
@@ -398,6 +398,26 @@ extension SnapshotTests {
       }
     }
 
+    @Test func upsertWithoutID_OtherConflict() {
+      assertQuery(
+        RemindersList.upsert(RemindersList.Draft(name: "Personal"))
+          .returning(\.self)
+      ) {
+        """
+        INSERT INTO "remindersLists"
+        ("id", "color", "name")
+        VALUES
+        (NULL, 4889071, 'Personal')
+        ON CONFLICT ("id") DO UPDATE SET "color" = "excluded"."color", "name" = "excluded"."name"
+        RETURNING "id", "color", "name"
+        """
+      }results: {
+        """
+        UNIQUE constraint failed: remindersLists.name
+        """
+      }
+    }
```

Comment:
This is a test that now passes but without the changes in this PR it does not. In order to show this off I added a unique index to the "name" column of "remindersLists".

The problem here is that I am inserting a list with a duplicate name but no ID specified. When it sees a conflict with the name it updates that record with the fields of the list, even though we had no ID specified in the draft.